From: Hans-Joachim Picht <hpicht@redhat.com>
Date: Tue, 20 Jan 2009 16:58:04 +0100
Subject: [s390] qeth: unnecessary support ckeck in sysfs route6
Message-id: 20090120155804.GB21694@redhat.com
O-Subject: [RHEL5 U4 PATCH 2/8] s390 - qeth: remove unnecessary support ckeck in sysfs route6
Bugzilla: 474469
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
============

Removing this check improves usability because you do not have to
set the device online to initially set the ipv6 routing option.

Bugzilla
=========

BZ 474469
https://bugzilla.redhat.com/show_bug.cgi?id=474469

Upstream status of the patch:
=============================

The patch is upstream as of git commit
cc181282fb2fa1af6d532f1333dd42af4814ff17
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=cc181282fb2fa1af6d532f1333dd42af4814ff17

Test status:
============

The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Please ACK.

With best regards,

	--Hans

diff --git a/drivers/s390/net/qeth_sys.c b/drivers/s390/net/qeth_sys.c
index 4a2c5d9..15c818f 100644
--- a/drivers/s390/net/qeth_sys.c
+++ b/drivers/s390/net/qeth_sys.c
@@ -430,9 +430,6 @@ qeth_dev_route6_show(struct device *dev, struct device_attribute *attr, char *bu
 	if (!card)
 		return -EINVAL;
 
-	if (!qeth_is_supported(card, IPA_IPV6))
-		return sprintf(buf, "%s\n", "n/a");
-
 	return qeth_dev_route_show(card, &card->options.route6, buf);
 }
 
@@ -444,13 +441,6 @@ qeth_dev_route6_store(struct device *dev, struct device_attribute *attr, const c
 	if (!card)
 		return -EINVAL;
 
-	if (!qeth_is_supported(card, IPA_IPV6)){
-		PRINT_WARN("IPv6 not supported for interface %s.\n"
-			   "Routing status no changed.\n",
-			   QETH_CARD_IFNAME(card));
-		return -ENOTSUPP;
-	}
-
 	return qeth_dev_route_store(card, &card->options.route6,
 			            QETH_PROT_IPV6, buf, count);
 }
